packages/origin-ui-core/src/components/Organization/OrganizationView.tsx

```diff
@@ -43,22 +48,45 @@ export function OrganizationView() {
 
     const classes = useStyles(useTheme());
 
+    const setValues = (organization, activeCountriesParsed) => {
+        setFormValues({
+            ...organization,
+            headquartersCountry: Countries.find(
+                (c) => c.id === organization.headquartersCountry
+            ).name,
+            activeCountries: Countries.filter((c) => activeCountriesParsed.includes(c.id))
+                .map((country) => country.name)
+                .join(', '),
+            country: Countries.find((c) => c.id === organization.country).name
+    });
+    }
+
     useEffect(() => {
-        const organization = userOffchain?.organization;
-        if (organization) {
-            const activeCountriesParsed: number[] = JSON.parse(organization.activeCountries);
-            setFormValues({
-                ...organization,
-                headquartersCountry: Countries.find(
-                    (c) => c.id === organization.headquartersCountry
-                ).name,
-                activeCountries: Countries.filter((c) => activeCountriesParsed.includes(c.id))
-                    .map((country) => country.name)
-                    .join(', '),
-                country: Countries.find((c) => c.id === organization.country).name
-            });
+
+        if (!params.id) {
+            
+            const organization = userOffchain?.organization;
+            
+            if (organization) {
+                const activeCountriesParsed: number[] = JSON.parse(organization.activeCountries);
+                setValues(organization, activeCountriesParsed)
         }
-    }, [userOffchain]);
+        } else {
+        
+            const getOrganization = async () => {
+
+                const organizationsList = await organizationClient.getAll();
+                const organization = organizationsList.filter(org => org.id === parseInt(params.id))[0]
+                
+                if (organization) {
+                        const activeCountriesParsed: number[] = JSON.parse(organization.activeCountries);
+                        setValues(organization, activeCountriesParsed)
+                }
+        }
+        getOrganization()
+    }
+    return (() => ac.abort())
```

Comment:
`AbortController` is experimental feature. And besides shouldn't AbortController.signal be injected in fetch?